picard_plugin_tools/__init__.py

```diff
@@ -1,23 +1,67 @@
 import ast
-import os
 import json
+import os
+import re
 import zipfile
 
 from hashlib import md5
 
+import click
+
 # The file that contains json data
 PLUGIN_FILE_NAME = "PLUGINS.json"
 
+URL_REGEX = r"https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)"
+VERSION_REGEX = r"(?:(\d+\.(?:\d+\.)*\d+))"
+
+
+class VersionString(click.ParamType):
+    name = "version_string"
+
+    def convert(self, value, param, ctx):
+        if not re.match(VERSION_REGEX, value):
+            self.fail('%s is not a valid version string' % value, param, ctx)
+        else:
+            return value
+
+
+class APIVersions(click.ParamType):
+    name = "api_versions"
 
-KNOWN_DATA = [
-    'PLUGIN_NAME',
-    'PLUGIN_AUTHOR',
-    'PLUGIN_VERSION',
-    'PLUGIN_API_VERSIONS',
-    'PLUGIN_LICENSE',
-    'PLUGIN_LICENSE_URL',
-    'PLUGIN_DESCRIPTION',
-]
+    def convert(self, value, param, ctx):
+        api_versions = value.split(",")
+        if not len(api_versions):
+            self.fail('%s are not valid api versions' % value, param, ctx)
+        for api_version in api_versions:
+            if not re.match(VERSION_REGEX, api_version):
+                self.fail('%s is not a valid API version' % api_version, param, ctx)
+        else:
+            return api_versions
+
+
+class URLString(click.ParamType):
+    name = "url_string"
+
+    def convert(self, value, param, ctx):
+        if not re.match(URL_REGEX, value):
+            self.fail('%s is not a valid URL' % value, param, ctx)
+        else:
+            return value
+
+
+VERSION_STRING = VersionString()
+API_VERSIONS = APIVersions()
+URL_STRING = URLString()
```

Comment:
I don't think those should be capitalized. I would prefer a more explicit `type_version_string` or even get rid of those variables since you may directly instanciate in the KNOW_DATA dict